packages/vats/src/virtual-purse.js

```diff
@@ -102,6 +105,22 @@ export const makeVirtualPurseKitIKit = (
  *   current balance iterable for a given brand.
  */
 
+/**
+ * Until https://github.com/Agoric/agoric-sdk/issues/9407 is fixed, this
+ * function restricts the `optAmountShape`, if provided, to be a concrete
+ * `Amount` rather than a `Pattern` as it is supposed to be.
+ *
+ * TODO: Once https://github.com/Agoric/agoric-sdk/issues/9407 is fixed, remove
+ * this function and all calls to it.
+ *
+ * @param {Pattern} [optAmountShape]
+ */
+const legacyRestrictAmountShapeArg = optAmountShape => {
+  if (optAmountShape && !matches(optAmountShape, AmountShape)) {
```

Comment:
Have we verified that `makeVirtualPurseKitIKit` is never called with an `amountShape` different than `undefined` or `AmountShape` ?

Reply:
I just looked. Yes. There are only two calls, and both have zero args, letting both params default.

Done.